From 979194c4a8e3bd25c99bb6970ef8d43bea648327 Mon Sep 17 00:00:00 2001
From: Takuto Ikuta <tikuta@google.com>
Date: Tue, 9 Mar 2021 19:00:02 +0900
Subject: [PATCH] Support major version suffix in Go modules

This recognizes package using major version suffix in github. It won't
be treated as subpackage.
---
 util/normalizename_test.go | 5 +++++
 util/util.go               | 2 +-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/util/normalizename_test.go b/util/normalizename_test.go
index d5a454b..df1b59f 100644
--- a/util/normalizename_test.go
+++ b/util/normalizename_test.go
@@ -15,6 +15,11 @@ func TestNormalizeName(t *testing.T) {
 			root:  "github.com/Masterminds/cookoo",
 			extra: "web/io/foo",
 		},
+		{
+			input: "github.com/dgraph-io/badger/v3/y",
+			root:  "github.com/dgraph-io/badger/v3",
+			extra: "y",
+		},
 		{
 			input: `github.com\Masterminds\cookoo\web\io\foo`,
 			root:  "github.com/Masterminds/cookoo",
diff --git a/util/util.go b/util/util.go
index 5a186b4..5e92e3a 100644
--- a/util/util.go
+++ b/util/util.go
@@ -210,7 +210,7 @@ type vcsInfo struct {
 var vcsList = []*vcsInfo{
 	{
 		host:    "github.com",
-		pattern: `^(?P<rootpkg>github\.com/[A-Za-z0-9_.\-]+/[A-Za-z0-9_.\-]+)(/[A-Za-z0-9_.\-]+)*$`,
+		pattern: `^(?P<rootpkg>github\.com/[A-Za-z0-9_.\-]+/[A-Za-z0-9_.\-]+(/v[0-9]+)?)(/[A-Za-z0-9_.\-]+)*$`,
 	},
 	{
 		host:    "bitbucket.org",
-- 
2.30.1.766.gb4fecdf3b7-goog

